### ISSUE TITLE
Content assist activates build because of auto file save

### ISSUE THREAD

**Issue by Liano** [RustDT/RustDT#64]
Hello,
I just installed RustDT, everything is configured as described in the user guide, the issue I am facing right now is whenever I trigger code completion using '.' or '::' the completion appears, but quickly causes cargo to build the project, remove completion context menu and activates Console window.
Am I missing something or is this a bug actually?
Below is Console output.

```
====================  Starting Rust build  ====================
Cleared problem markers for test.
************  Building Rust project: test  ************
>> Running: C:\portable_apps\rust_1.2\bin\cargo.exe build
An unknown error occurredBuild failed, waiting for other jobs to finish...


To learn more, run the command again with --verbose.
  ^^^ Terminated, exit code: 101 ^^^
>> Running: C:\portable_apps\rust_1.2\bin\cargo.exe test --no-run
   Compiling hello_world v0.0.1 (file:///C:/Users/CraCKerO/workspace2/test)
src\main.rs:8:6: 8:8 error: unresolved name `io`
src\main.rs:8     }io
                   ^~
error: aborting due to previous error
Could not compile `hello_world`.

To learn more, run the command again with --verbose.
  ^^^ Terminated, exit code: 101 ^^^
************  Build terminated.  ************
```

BR.

**Comment by Koka** [comment]
I have the same problem. I use the following workaround: don't keep console minimized. If console is restored to full open state build process will not steal focus of your editor and content assist window will not disappear.

**Comment by bruno-medeiros** [comment]
You can also disable the "Project / Build automatically" option. What is happening here, is that when completion is invoked (whether explicitly or implicitly), the current file will be saved. The save triggers a build if "Build automatically" is enable.

I will if I can try to fix this so that invoking completion doesn't require saving the file.

**Comment by bruno-medeiros** [comment]
You can also disable the "Project / Build automatically" option. What is happening here, is that when completion is invoked (whether explicitly or implicitly), the current file will be saved. The save triggers a build if "Build automatically" is enabled.

I will see if I can try to fix this so that invoking completion doesn't require saving the file.

**Comment by Liano** [comment]
Thank you all for your answer,
@Koka I tried the workaround you suggested but it doesn't work, even if the the console is not minimized CA menu disappears
@bruno-medeiros By disabling the auto-build option it now works fine.
Thank you.

**Comment by bruno-medeiros** [comment]
I still wanna see if I can make RustDT handle this better, when "Build automatically" is enabled

**Comment by bruno-medeiros** [comment]
> I still wanna see if I can make RustDT handle this better, when "Build automatically" is enabled
This has now be addressed properly in 0.4.1 -> the rust builder is no longer invoked automatically when a file is save, even if "Build automatically" is enabled.

**Comment by bruno-medeiros** [comment]
> I still wanna see if I can make RustDT handle this better, when "Build automatically" is enabled

This has now be addressed properly in 0.4.1 -> the rust builder is no longer invoked automatically when a file is save, even if "Build automatically" is enabled.

**Comment by Liano** [comment]
Thank you for your efforts!
